runtime-modules/storage/src/tests/data_directory.rs

```diff
@@ -46,13 +47,352 @@ fn add_content_fails_with_invalid_origin() {
+        assert_eq!(res, Err(Error::<Test>::QuotaObjectsLimitExceeded.into()));
+    });
+}
+
+#[test]
+fn add_content_global_size_limit_reached() {
+    let global_quota_size_limit = 0;
+    let global_quota_objects_limit = 50;
+
+    ExtBuilder::default()
+        .global_quota(Quota::new(
+            global_quota_size_limit,
+            global_quota_objects_limit,
+        ))
+        .build()
+        .execute_with(|| {
+            let sender = 1u64;
+
+            let owner = StorageObjectOwner::Member(1u64);
+
+            let content_parameters = ContentParameters {
+                content_id: 1,
+                type_id: 1234,
+                size: global_quota_size_limit + 1,
+                ipfs_content_id: vec![1, 2, 3, 4],
+            };
+
+            // Make an attempt to register a content, when uploading is blocked.
+            let res = TestDataDirectory::add_content(
+                Origin::signed(sender),
+                owner,
+                vec![content_parameters],
+            );
+            assert_eq!(res, Err(Error::<Test>::GlobalQuotaSizeLimitExceeded.into()));
+        });
+}
+
+#[test]
+fn add_content_global_objects_limit_reached() {
+    let global_quota_size_limit = 50000;
+    let global_quota_objects_limit = 0;
+
+    ExtBuilder::default()
+        .global_quota(Quota::new(
+            global_quota_size_limit,
+            global_quota_objects_limit,
+        ))
+        .build()
+        .execute_with(|| {
+            let sender = 1u64;
+
+            let owner = StorageObjectOwner::Member(1u64);
+
+            let content_parameters = ContentParameters {
+                content_id: 1,
+                type_id: 1234,
+                size: 0,
+                ipfs_content_id: vec![1, 2, 3, 4],
+            };
+
+            // Make an attempt to register a content, when uploading is blocked.
+            let res = TestDataDirectory::add_content(
+                Origin::signed(sender),
+                owner,
+                vec![content_parameters],
+            );
+            assert_eq!(
+                res,
+                Err(Error::<Test>::GlobalQuotaObjectsLimitExceeded.into())
+            );
+        });
+}
+
+#[test]
+fn delete_content() {
+    with_default_mock_builder(|| {
+        let sender = 1u64;
+
+        let owner = StorageObjectOwner::Member(1u64);
+
+        let content_id = 1;
+
+        let content_parameters = ContentParameters {
+            content_id,
+            type_id: 1234,
+            size: 1,
+            ipfs_content_id: vec![1, 2, 3, 4],
+        };
+
+        // Register a content with 1234 bytes of type 1, which should be recognized.
+
+        TestDataDirectory::add_content(
+            Origin::signed(sender),
+            owner.clone(),
+            vec![content_parameters],
+        )
+        .unwrap();
+
+        let res =
+            TestDataDirectory::remove_content(Origin::signed(sender), owner, vec![content_id]);
+
+        assert!(res.is_ok());
```

Comment:
and a specific macro for testing success:
```rust
assert_ok!(res);
```

I think macro also takes optional second argument (a string) for a human friendly message on assertion failure.